### PR TITLE
i18n(de): update translation reference/configuration

### PR DESCRIPTION
**Pull request by trueberryless**
#### Description

Translation of `reference/configuration.mdx` into German language.

### CODE REVIEW

**Comment by trueberryless** [comment]
Link fixed in #2458

**Comment by trueberryless** [comment]
Link fixed in #2464

File: docs/src/content/docs/de/reference/configuration.mdx [hunk 1]
```diff
@@ -131,7 +153,7 @@ Autogenerierte Untergruppen respektieren standardmäßig die Eigenschaft `collap
 sidebar: [
   // Eine zusammengefasste Gruppe von Links
   {
-    label: 'Collapsed Links',
+    label: 'Zusammengeklappte Links',
     collapsed: true,
     items: [
```
Comment (HiDeoo) [inline docs/src/content/docs/de/reference/configuration.mdx:158]: This example seems to be missing the changes from https://github.com/withastro/starlight/pull/1874/files#diff-889e6ae5a36c15d03a30012e2936df94a63753b335bd5a8c56fac9a42e5c1dff and also proper highlights.

File: docs/src/content/docs/de/reference/configuration.mdx [hunk 2]
```diff
@@ -398,7 +527,7 @@ Eine Seite kann diese Einstellung oder den Linktext und/oder die URL mit Hilfe d
 ### `favicon`
 
 **Typ:** `string`  
-**Standard:** `'/favicon.svg'`
+**Voreinstellung:** `'/favicon.svg'`
```
Comment (HiDeoo) [inline docs/src/content/docs/de/reference/configuration.mdx:530]: I cannot suggest a change for it as it's outside of the review range but the second example in this section is missing the change from https://github.com/withastro/starlight/pull/1627/files#diff-889e6ae5a36c15d03a30012e2936df94a63753b335bd5a8c56fac9a42e5c1dff

**Review by HiDeoo — CHANGES_REQUESTED** [review]
Thanks for the update :raised_hands:

As a non-native speaker, I reviewed this PR using the [Docs Translation Review Checklist](https://contribute.docs.astro.build/reviewers/reviewing-translations/). Left a few comments regarding some missing changes but otherwise, this looks good to me.

**Review by HiDeoo — APPROVED** [review]
Thanks for updating the PR, everything looks good now :rocket: I'll temporarily merge with broken links so they can be fixed in the next PRs.

**Pull request merged**